lib/screens/v2/widget_instance/reconstructed_alert.ex

```diff
@@ -90,14 +94,32 @@ defmodule Screens.V2.WidgetInstance.ReconstructedAlert do
       |> hd()
 
     if is_nil(direction_id) do
-      nil
+      informed_stop_ids = Enum.into(informed_entities, MapSet.new(), & &1.stop)
+
+      :screens
+      |> Application.get_env(:reconstructed_alert_headsign_matchers)
+      |> Map.get(stop_id)
+      |> Enum.find_value(nil, fn {informed, not_informed, headsign} ->
+        if alert_region_match?(to_set(informed), to_set(not_informed), informed_stop_ids),
+          do: headsign,
+          else: false
+      end)
     else
       @route_directions
       |> Map.get(route_id)
       |> Enum.at(direction_id)
     end
   end
 
+  defp to_set(stop_id) when is_binary(stop_id), do: MapSet.new([stop_id])
+  defp to_set(stop_ids) when is_list(stop_ids), do: MapSet.new(stop_ids)
+  defp to_set(%MapSet{} = already_a_set), do: already_a_set
```

Comment:
I don't think this case is necessary. Passing in a MapSet to `MapSet.new` will return a MapSet. You should be able to combine this and the clause above it by removing the guard.

```suggestion
  defp to_set(stop_ids), do: MapSet.new(stop_ids)
```

Reply:
Nice!